crawl-ref/source/dat/des/branches/zot.des

```diff
@@ -2786,6 +2786,41 @@ MAP
 ............?............
 ENDMAP
 
+NAME:   dreamdust_zin_vs_xom
+TAGS:   no_monster_gen
+ORIENT: float
+DEPTH:  Zot, !Zot:$
+COLOUR: v = white
+TILE:   v = dngn_metal_wall_white
+FTILE:  _`1 = floor_w_marble
+KFEAT:  _ = altar_zin
+TILE:   G = dngn_statue_angel
+MONS:   angel god:zin dbname:zin_angel, \
+        killer klown-shaped pillar of salt, \
+        killer klown, \
+        butterfly, \
+        chaos spawn, \
+        glowing shapeshifter
+KFEAT:  O = altar_xom
+KFEAT:  ^ = dispersal trap
+NSUBST: - = 4=2 / 4=2- / -
+NSUBST: ? = O / 2=^ / 2=3 / 2=3? / 2=4 / ?
+SUBST:  ? = ":200 W:100 n:60 x:40 O:40 v b ^ 4 5 6 T U V Y
```

Comment:
Although the chance for it is pretty low in this particular case, you want to avoid a substitution that can create teleport closets. The best way to do this is to use a checkerboard pattern for your glyph:
```
---"?"?"?"x
---?"?"?"?x
---"?"?"?"x
---?"?"?"?x
---"?"?"?"x
```
You may want to modify you substitution accordingly. Since much of the `?` will still become floor, it will still look randomized, just not quite as much as it would with the uniform block of `?`. You can get a bit fancier and try to add some "seeds" of specific terrain or a second interspersed substitution pattern, but you just have to ensure that your final result can never make teleport closets. Masking the substitution area with e.g. `no_tele_into` and other tags isn't an alternative to this, since we want to avoid hidden disconnected zones. If the disconnected zones are visible (i.e. through glass) and masked appropriately to not allow teleport and not generate items/monsters, then it's fine.